gradio/oauth.py

```diff
@@ -124,19 +123,50 @@ def _add_mocked_oauth_routes(app: fastapi.FastAPI) -> None:
     @app.get("/login/huggingface")
     async def oauth_login(request: fastapi.Request):
         """Fake endpoint that redirects to HF OAuth page."""
-        return RedirectResponse("/login/callback")
+        # Define target (where to redirect after login)
+        redirect_uri = _generate_redirect_uri(request)
+        return RedirectResponse(
+            "/login/callback?" + urllib.parse.urlencode({"_target_url": redirect_uri})
+        )
 
     @app.get("/login/callback")
     async def oauth_redirect_callback(request: fastapi.Request) -> RedirectResponse:
         """Endpoint that handles the OAuth callback."""
         request.session["oauth_info"] = mocked_oauth_info
-        return RedirectResponse("/")
+        return _redirect_to_target(request)
 
     @app.get("/logout")
     async def oauth_logout(request: fastapi.Request) -> RedirectResponse:
         """Endpoint that logs out the user (e.g. delete cookie session)."""
         request.session.pop("oauth_info", None)
-        return RedirectResponse("/")
+        return _redirect_to_target(request)
+
+
+def _generate_redirect_uri(request: fastapi.Request) -> str:
+    if "_target" in request.query_params:
+        # if `_target` already in query params => respect it
+        target = request.query_params["_target_url"]
```

Comment:
IIUC this should be:
```suggestion
    if "_target_url" in request.query_params:
        # if `_target_url` already in query params => respect it
        target = request.query_params["_target_url"]
```